src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.BalanceOperationHandler;
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.operation.PurchaseOperationHandler;
+import core.basesyntax.operation.ReturnOperationHandler;
+import core.basesyntax.operation.SupplyOperationHandler;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SEPARATOR = ",";
+    private static final String FIRST_LINE_IN_OUTPUT = "fruit,quantity" + System.lineSeparator();
+    private OperationStrategy operationStrategy;
+
+    @Override
+    public String createFinalReport(Map<String, Integer> fruitsMap) {
+        StringBuilder builder = new StringBuilder(FIRST_LINE_IN_OUTPUT);
+        for (Map.Entry<String, Integer> map : fruitsMap.entrySet()) {
+            builder.append(map.getKey());
+            builder.append(SEPARATOR);
+            builder.append(map.getValue());
+            builder.append(System.lineSeparator());
+        }
+        return builder.toString();
+    }
+
+    @Override
+    public void editMapReportDependingOnOperation(List<FruitTransaction> fruits) {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN,
+                new ReturnOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyOperationHandler());
```

Comment:
Please remove hardcoded map upgrade, move to class field and accept map as argument in constructor.
Create proper map in Main class and pass to constructor of service.
With current implementation you recreate map each time method is called